rivertile/main.zig

```diff
@@ -179,6 +180,37 @@ const Output = struct {
                                 if (arg >= 1) output.main_count = @intCast(arg);
                             },
                         }
+
+                        var mainCount: [32:0]u8 = undefined;
+                        _ = std.fmt.bufPrintZ(&mainCount, "{d}", .{output.main_count}) catch {
+                            std.log.err("could not write {d} to buf", .{output.main_count});
+                            return;
+                        };
+
+                        const args = &[_][*:0]const u8{
+                            "set-main-count",
+                            &mainCount,
+                        };
+
+                        riverctl.run(args) catch |err| {
+                            switch (err) {
+                                error.ConnectFailed => {
+                                    std.log.err("failed to connect to wayland compositor", .{});
+                                },
+                                error.RiverControlNotAdvertised => {
+                                    std.log.err("river control not advertised", .{});
+                                },
+                                error.SeatNotAdverstised => {
+                                    std.log.err("seat not advertised", .{});
+                                },
+                                error.CannotGetRegistry => {
+                                    std.log.err("failed to get registry", .{});
+                                },
+                                error.RunCommandFailed => {
+                                    std.log.err("failed to run command", .{});
+                                },
+                            }
+                        };
```

Comment:
I am pretty sure this is out of scope for rivertile and will not be merged.

Rivertile is supposed to be a reference implementation with a very restricted feature set and has no ambitions to serve the use cases of a wide range of users. Users running (and creating) other layout generators is highly encouraged.